java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -106,6 +127,23 @@ private void visitLambdaExpression(LambdaExpressionTree tree) {
     return Optional.empty();
   }
 
+  private static Optional<String> getTypeCast(@Nullable Tree statement, LambdaExpressionTree tree) {
+    if (statement == null) {
+      return Optional.empty();
+    }
+    Tree expr = statement;
+    if (expr.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+      expr = ExpressionUtils.skipParentheses((ParenthesizedTree) statement);
+    }
+    if (expr.is(Tree.Kind.TYPE_CAST)) {
+      TypeCastTree typeCastTree = (TypeCastTree) expr;
+      if (isSingleParamCast(typeCastTree.expression(), tree)) {
+        return Optional.of(typeCastTree.type().symbolType().name());
```

Comment:
what if the type is unknown but the symbols are matched by the frontend? I fear it's going to return "!unknown!" here.

Instead of:
```java
return Optional.of(typeCastTree.type().symbolType().name());
```
I would use something like so we use directly the TypeTree, which, even unresolved, will display the correct name:
```java
return Optional.of(ExpressionsHelper.concatenate(typeCastTree.type()));
```